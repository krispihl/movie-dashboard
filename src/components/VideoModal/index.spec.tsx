import { VideoModal } from "./index";
import { render } from '@testing-library/react';

jest.mock('../../contexts/movie-details', () => ({
    useMovieDetails: () => jest.fn()
}));

describe('src/components/VideoModal', () => {
    it('should render VideoModal component correctly', async () => {
        const { container } = render(<VideoModal embedId='some_youtube_vid_id' />);
        const iframeEl = document.querySelector("[title='Embedded youtube video']");

        expect(container).toMatchSnapshot();
        expect(iframeEl).toBeInTheDocument();
        expect(iframeEl).toHaveAttribute('src', 'https://www.youtube.com/embed/some_youtube_vid_id?autoplay=1');
    });
});