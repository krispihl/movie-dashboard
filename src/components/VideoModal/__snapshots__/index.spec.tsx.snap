// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/VideoModal should render VideoModal component correctly 1`] = `
<div>
  <div
    class="video"
  >
    <div
      class="video__container"
    >
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        src="https://www.youtube.com/embed/some_youtube_vid_id?autoplay=1"
        title="Embedded youtube video"
      />
    </div>
  </div>
</div>
`;
